src/main/java/uk/gov/companieshouse/web/accounts/controller/cic/CicCompanyDetailController.java

```diff
@@ -0,0 +1,93 @@
+package uk.gov.companieshouse.web.accounts.controller.cic;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import uk.gov.companieshouse.web.accounts.annotation.NextController;
+import uk.gov.companieshouse.web.accounts.controller.BaseController;
+import uk.gov.companieshouse.web.accounts.exception.ServiceException;
+import uk.gov.companieshouse.web.accounts.model.company.CompanyDetail;
+import uk.gov.companieshouse.web.accounts.service.company.CompanyService;
+
+import javax.servlet.http.HttpServletRequest;
+import java.time.LocalDate;
+
+@Controller
+@NextController(CICStepsToCompleteController.class)
+@RequestMapping("/accounts/company/{companyNumber}/cic/details")
+public class CicCompanyDetailController extends BaseController {
+
+    @Autowired
+    CompanyService companyService;
+
+    @Override
+    protected String getTemplateName() {
+        return "company/companyDetail";
+    }
+
+    private static final String TEMPLATE_HEADING_IS_CIC = "Confirm company details";
+
+    private static final String TEMPLATE_HEADING_ACC_NOT_DUE = "This CIC’s annual accounts are not due yet";
+    private static final String TEMPLATE_LINK_ACC_TEXT = "File CIC report and accounts for a different CIC";
+    private static final String TEMPLATE_LINK_ACC_URL = "/company-lookup/search?forward=%2Faccounts%2Fcompany%2F%7BcompanyNumber%7D%2Fcic%2Fdetails";
+
+    private static final String TEMPLATE_HEADING_NOT_CIC = "This is not a Community Interest Company";
+    private static final String TEMPLATE_LINK_NOT_CIC_TEXT = "File accounts for other company types";
+    private static final String TEMPLATE_LINK_NOT_CIC_URL = "https://www.gov.uk/file-your-company-annual-accounts";
+
+    private static final String MODEL_ATTR_HEADING = "templateHeading";
+    private static final String MODEL_ATTR_LINK_TEXT = "templateLinkText";
+    private static final String MODEL_ATTR_LINK_URL = "templateLinkUrl";
+
+    private static final boolean SHOW_CONTINUE_BUTTON = true;
+    private static final String MODEL_ATTR_HIDE_CONTINUE_BUTTON = "showContinueButton";
```

Comment:
Ideally we wouldn't reference 'button' in the controller; should the format of the screen ever change and we used something other than a button, we would have to update this reference in the controller too